lib/core/elastic_connector_actions.rb

```diff
@@ -114,8 +114,30 @@ def complete_sync(connector_package_id, job_id, status)
         end
       end
 
-      def client
-        @client ||= Utility::EsClient.new
+      def fetch_document_ids(index_name)
+        result = []
+        page_size = 1000
+        pit_id = client.open_point_in_time(:index => index_name, :keep_alive => '1m', :expand_wildcards => 'all')['id']
+        body = {
+          :query => { :match_all => {} },
+          :sort => [{ :id => { :order => :asc } }],
+          :pit => {
+            :id => pit_id,
+            :keep_alive => '1m'
+          },
+          :size => page_size,
+          :_source => false
+        }
+        loop do
+          response = client.search(:body => body)
+          hits = response['hits']['hits']
+          ids = hits.map { |h| h['_id'] }
+          result += ids
+          body[:search_after] = hits.last['sort']
```

Comment:
I did not test it. But from the doc, to get the next page, you should use
```
body[:pit][:id] = response['pit_id']
```

instead of
```
body[:search_after] = hits.last['sort']
```

Because you use PIT when searching

Reply:
I think this requests all use PIT, it just looks like they do not. It's initialised one time in `body` variable and then each new request modifies original object (`body[:search_after] = hits.last['sort']`) and sends original request with original params again:

```ruby
body = {
          :query => { :match_all => {} },
          :sort => [{ :id => { :order => :asc } }],
          :pit => {
            :id => pit_id,
            :keep_alive => '1m'
          },
          :size => page_size,
          :_source => false
        }
```

Reply:
According to the doc, if you are using PIT, instead of having `search_after` in the next request body, you should set `pit.id` to the `pit_id` of the response.

Reply:
Indeed, but PIT was generated beforehands and it won't change between requests, or is this case possible?

Reply:
This threw me for a loop too. From [these docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html)

> To get the next page of results, rerun the previous search using the last hit’s sort values (including the tiebreaker) as the search_after argument. If using a PIT, use the latest PIT ID in the pit.id parameter. The search’s query and sort arguments must remain unchanged. If provided, the from argument must be 0 (default) or -1.

it does say that you need to take the previous response's PIT and use it in the next request. It does not say that the PIT will remain unchanged, though I thought it would? But "latest PIT ID" makes me worry that this is subject to change. Maybe worth updating this to just explicitly re-set it each time?